File: Advanced Programming Methods/Toy Language C#/A2/Model/Expressions/Expression.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace A2.Model.Expressions
{
    public abstract class Expression
    {
        public abstract int eval(Model.Data_Structures.IDictionary<string, int> dict);
        public abstract override string ToString();
    }

}

File: Advanced Programming Methods/Toy Language JavaFX/src/Model/Expressions/Expression.java
package Model.Expressions;

import Model.DataStructures.IDictionary;
import Model.DataStructures.IHeap;
import Model.DataStructures.MyException;

public abstract class Expression {
    public abstract int eval(IDictionary<String, Integer> dict, IHeap<Integer, Integer> heap) throws MyException;
    public abstract String toString();
}
